matrix_client/api.py

```diff
@@ -615,3 +616,65 @@ def get_room_members(self, room_id):
         """
         return self._send("GET", "/rooms/{}/members".format(quote(room_id)),
                           api_path=MATRIX_V2_API_PATH)
+
+    def get_devices(self):
+        """
+            Gets information about all devices for the current user
+        """
+        return self._send("GET", "/devices",
+                          api_path=MATRIX_UNSTABLE_API_PATH)
+
+    def get_device(self, device_id):
+        """
+            Gets information on a single device, by device id.
+        """
+        return self._send("GET", "/devices/%s" % device_id,
+                          api_path=MATRIX_UNSTABLE_API_PATH)
+
+    def update_device_info(self, device_id, display_name):
+        """ Update the display name of a device.
+        Args:
+            device_id (str): The device ID of the device to update.
+            display_name (str): New display name for the device.
+        """
+        content = {
+            "display_name": display_name
+        }
+        return self._send(
+            "PUT",
+            "/devices/%s" % device_id,
+            content=content,
+            api_path=MATRIX_UNSTABLE_API_PATH)
+
+    def delete_device(self, auth_body, device_id):
+        """ Deletes the given device, and invalidates any access token assoicated with it.
+        Requires Auth.
+        Args:
+            auth_body (dict): Authentication params.
+            device_id (str): The device ID of the device to delete.
+        """
+        content = {
+            "auth": auth_body
+        }
+        return self._send(
+            "DELETE",
+            "/devices/%s" % device_id,
+            content=content,
+            api_path=MATRIX_UNSTABLE_API_PATH)
+
+    def delete_devices(self, auth_body, devices):
```

Comment:
In practice, I think this only accepts `m.login.password` type auth https://github.com/matrix-org/synapse/blob/develop/synapse/rest/client/v2_alpha/devices.py#L133 but that's not specified in the spec (so maybe still subject to change?) 

https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#delete-matrix-client-unstable-devices-deviceid